### DISCUSSION TITLE
Renogy OEM RS485 to USB Serial Cable - Works but disconnects randomly

### DISCUSSION THREAD

**Discussion by TimGFoley** [Louisvdw/dbus-serialbattery#235] [General]
Has anyone tested the Renogy OEM RS485 to USB Serial Cable with this driver?

https://ca.renogy.com/rs485-to-usb-serial-cable/

I just ordered one to try it out but was curious if anyone had already tried.

**Comment by TimGFoley** [comment]
Well I can answer my own question.  USB/485 cable arrived today from Renogy.  Intalled driver manually via SSH and rebooted.

Connected cable to battery and to Venus GX, after a few moments "poof" SerialBattery(Renogy) v0.13 appeared and then promptly disconnected.   I had the Renogy display attached so I removed that and reconnected the RS485 cable to that same port.   Serial Battery came back after a few moments.

I've been watching it for a few minutes and the battery did disconnect a second time but then reconnected.  I'll keep an eye on it but the Renogy cable worked.

It disconnected yet again when I tried to view details so maybe not 100% stable yet.

![image](https://user-images.githubusercontent.com/5216131/194171022-bed84c44-94a4-41c7-ba3a-ed75d31d6a07.png)

I just left it at the details page and eventually it came back online so I could see that page as well:

![image](https://user-images.githubusercontent.com/5216131/194171739-c0255e9e-097e-4e4b-9078-82755f1fff61.png)

Parameters Charge Current Limit is rather low but I'll do some reading before I play with these:

![image](https://user-images.githubusercontent.com/5216131/194172029-4cc629af-9a4e-4c82-afe8-35f049beffce.png)

**Reply by TimGFoley** [reply]
Checking the FAQ and Troubleshooting, log looks good, no errors showing up when battery disconnects:

![image](https://user-images.githubusercontent.com/5216131/194172678-c61e8d0a-38fa-4921-a012-b78485b56684.png)

Will keep digging.

EDIT: Caught the log when disconnected:

![image](https://user-images.githubusercontent.com/5216131/194173778-987d5437-a44c-4fdb-90ca-b0b38a83af07.png)

**Reply by Louisvdw** [reply]
It looks like the BMS could not be read in that instance and the driver did not handle that error well enough. I'll add some extra checks in to help.

**Reply by TimGFoley** [reply]
Thanks Louis, mine is always connected so I can test anything you need.   Awesome job on this by the way!

**Reply by stanhausc** [reply]
> It looks like the BMS could not be read in that instance and the driver did not handle that error well enough. I'll add some extra checks in to help.

This also happens here, but I don't get any error messages.

![image](https://user-images.githubusercontent.com/85182432/204266727-8f311c31-4ffe-4840-b032-f238bedf6f29.png)
